oslabs-beta/reactive
# fix/webpackBundling

## Linked issue/ticket: #66

## Description
This PR introduces two main changes:

1. **Test Suite Addition:**
   - Implemented beginningtest suite for the Dendrogram component using JSDOM
   - Added tests to verify SVG structure and rendering
   - Created tests for tree structure data validation
   - Configured test runner to include dendrogram.test.js 

2. **Webpack Build Fix:**
   - Resolved dendrogram not being updated after build
       - Current quick fix removed pre-build as linters silent error stops build
   - Implemented proper cleaning strategy for build outputs:
     - Maintained `clean: true` for extension builds / to be explicit for other devs
     - Set `clean: false` for webview builds / to be explicit for other devs


These changes improve code reliability through testing and resolve build conflicts that were causing dendrogram to not be rendered

## Reproduction steps
To test the new test suite:
1. Run `npm install` to ensure test dependencies are installed
2. Execute `npm test` to run the full test suite
3. Verify that dendrogram.test.js passes all tests
4. Check test coverage for the Dendrogram component

To verify webpack fix:
1. Delete the dist folder
2. Run `npm run build`
3. Verify that both extension.js and webview.js are present in dist
4. Run build again to confirm no "file already exists" errors

## Checklist
- [x] I've followed the [Contributing guidelines](https://github.com/oslabs-beta/.github/blob/main/docs/CONTRIBUTING.md)
- [x] I've titled my PR according to the [Conventional Commits spec](https://conventionalcommits.org)
- [] I've added tests that fail without this PR but pass with it
